### PR TITLE
Enable TRT EP for C#

### PR DESCRIPTION
**Pull request by chilo-ms**
**Description**:
Enable TRT EP for C#. 
This PR is derived from #7179 but without supporting provider options configuration.

### CODE REVIEW

**Comment by jywu-msft** [comment]
wonder why CI isn't automatically kicked off. are you missing some membership

File: csharp/src/Microsoft.ML.OnnxRuntime/SessionOptions.cs [hunk 1]
```diff
@@ -75,6 +76,22 @@ public static SessionOptions MakeSessionOptionWithCudaProvider(int deviceId = 0)
...
+        /// </summary>
+        /// <param name="deviceId"></param>
+        /// <returns>A SessionsOptions() object configured for execution on deviceId</returns>
+        public static SessionOptions MakeSessionOptionWithTensorrtProvider(int deviceId = 0)
+        {
+            CheckTensorrtExecutionProviderDLLs();
+            SessionOptions options = new SessionOptions();
+            NativeApiStatus.VerifySuccess(NativeMethods.OrtSessionOptionsAppendExecutionProvider_Tensorrt(options.Handle, deviceId));
+            NativeApiStatus.VerifySuccess(NativeMethods.OrtSessionOptionsAppendExecutionProvider_CUDA(options.Handle, deviceId));
+            NativeApiStatus.VerifySuccess(NativeMethods.OrtSessionOptionsAppendExecutionProvider_CPU(options.Handle, 1));
```
Comment (yuslepukhin) [inline csharp/src/Microsoft.ML.OnnxRuntime/SessionOptions.cs:91]: We are not guarding against the options leak on exception here.

Comment (jywu-msft) [reply csharp/src/Microsoft.ML.OnnxRuntime/SessionOptions.cs:91]: looks like all the other MakeSessionOption static methods in this file are following the same pattern.
what's your suggestion. how should we recover here. this method returns options.
prevent the leak but propagate the exception?

Comment (yuslepukhin) [reply csharp/src/Microsoft.ML.OnnxRuntime/SessionOptions.cs:91]: We just try/catch. In catch we do options.Dispose() and then rethrow.

Comment (yuslepukhin) [reply csharp/src/Microsoft.ML.OnnxRuntime/SessionOptions.cs:91]: Need to fix the other location

Comment (chilo-ms) [reply csharp/src/Microsoft.ML.OnnxRuntime/SessionOptions.cs:91]: @yuslepukhin are you suggesting using try/catch as the previous discussion?  https://github.com/microsoft/onnxruntime/pull/7315#discussion_r613934184

Comment (chilo-ms) [reply csharp/src/Microsoft.ML.OnnxRuntime/SessionOptions.cs]: I've added try/catch around native methods.

Comment (chilo-ms) [reply csharp/src/Microsoft.ML.OnnxRuntime/SessionOptions.cs]: If you think others should also properly address this options leak, I can help modify in another PR.

**Review by jywu-msft — APPROVED** [review]

**Pull request merged**